Unable to instantiate ContextResolver when creating integration

```
tus change handler
2017-08-28 07:29:15.961  INFO [-,,,] 1 --- [pool-4-thread-1] i.s.c.integration.IntegrationController  : Integration 5 : Start processing integration with ActivateHandler
2017-08-28 07:29:15.961  INFO [-,,,] 1 --- [pool-4-thread-1] i.s.c.integration.IntegrationController  : Integration 5 : Setting status to Pending
2017-08-28 07:29:30.099  WARN [-,eb8a1bfc8c8a4d7f,eb8a1bfc8c8a4d7f,false] 1 --- [  XNIO-3 task-3] i.s.r.v.h.tests.TestSupportHandler       : user 8c1dc57a-90fc-4571-bbd9-fac8dbcad43f is resetting DB
2017-08-28 07:29:30.099  WARN [-,eb8a1bfc8c8a4d7f,eb8a1bfc8c8a4d7f,false] 1 --- [  XNIO-3 task-3] i.s.r.v.h.tests.TestSupportHandler       : user 8c1dc57a-90fc-4571-bbd9-fac8dbcad43f is deleting all integration deploymets
2017-08-28 07:29:57.126  INFO [-,,,] 1 --- [pool-4-thread-1] i.s.c.integration.IntegrationController  : Integration -Ksbjn_kzGflCW57ed4Q : Desired status "Activated" != current status "Pending" --> calling status change handler
2017-08-28 07:29:57.126  INFO [-,,,] 1 --- [pool-4-thread-1] i.s.c.integration.IntegrationController  : Integration -Ksbjn_kzGflCW57ed4Q : Start processing integration with ActivateHandler
2017-08-28 07:29:57.135 ERROR [-,,,] 1 --- [pool-4-thread-1] i.s.c.i.online.ActivateHandler           : Integration -Ksbjn_kzGflCW57ed4Q : Failure

java.lang.RuntimeException: java.lang.RuntimeException: RESTEASY003920: Unable to instantiate ContextResolver
	at org.jboss.resteasy.plugins.providers.RegisterBuiltin.register(RegisterBuiltin.java:41) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.getProviderFactory(ResteasyClientBuilder.java:334) ~[resteasy-client-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.register(ResteasyClientBuilder.java:444) ~[resteasy-client-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder.register(ResteasyClientBuilder.java:35) ~[resteasy-client-3.1.4.Final.jar!/:3.1.4.Final]
	at io.syndesis.core.Tokens.fetchProviderTokenFromKeycloak(Tokens.java:120) ~[core-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at io.syndesis.github.backend.KeycloakProviderTokenAwareGitHubClient.configureRequest(KeycloakProviderTokenAwareGitHubClient.java:48) ~[github-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at org.eclipse.egit.github.core.client.GitHubClient.createConnection(GitHubClient.java:272) ~[org.eclipse.egit.github.core-2.1.5.jar!/:na]
	at org.eclipse.egit.github.core.client.GitHubClient.createGet(GitHubClient.java:283) ~[org.eclipse.egit.github.core-2.1.5.jar!/:na]
	at org.eclipse.egit.github.core.client.GitHubClient.get(GitHubClient.java:729) ~[org.eclipse.egit.github.core-2.1.5.jar!/:na]
	at org.eclipse.egit.github.core.service.UserService.getUser(UserService.java:95) ~[org.eclipse.egit.github.core-2.1.5.jar!/:na]
	at io.syndesis.github.GitHubServiceImpl.getApiUser(GitHubServiceImpl.java:86) ~[github-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at io.syndesis.controllers.integration.online.ActivateHandler.getGitHubUser(ActivateHandler.java:194) ~[controllers-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at io.syndesis.controllers.integration.online.ActivateHandler.execute(ActivateHandler.java:105) ~[controllers-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at io.syndesis.controllers.integration.IntegrationController.lambda$callStatusChangeHandler$10(IntegrationController.java:173) [controllers-0.1-SNAPSHOT.jar!/:0.1-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_131]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
Caused by: java.lang.RuntimeException: RESTEASY003920: Unable to instantiate ContextResolver
	at org.jboss.resteasy.spi.ResteasyProviderFactory.processProviderContracts(ResteasyProviderFactory.java:1509) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.spi.ResteasyProviderFactory.registerProvider(ResteasyProviderFactory.java:1333) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.spi.ResteasyProviderFactory.registerProvider(ResteasyProviderFactory.java:1285) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.plugins.providers.RegisterBuiltin.registerProviders(RegisterBuiltin.java:77) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.plugins.providers.RegisterBuiltin.register(RegisterBuiltin.java:37) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	... 18 common frames omitted
Caused by: java.lang.IllegalArgumentException: RESTEASY003900: Unable to find a public constructor for provider class io.syndesis.runtime.ValidatorContextResolver
	at org.jboss.resteasy.spi.ResteasyProviderFactory.createConstructorInjector(ResteasyProviderFactory.java:2026) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.spi.ResteasyProviderFactory.createProviderInstance(ResteasyProviderFactory.java:2015) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.spi.ResteasyProviderFactory.addContextResolver(ResteasyProviderFactory.java:1027) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	at org.jboss.resteasy.spi.ResteasyProviderFactory.processProviderContracts(ResteasyProviderFactory.java:1503) ~[resteasy-jaxrs-3.1.4.Final.jar!/:3.1.4.Final]
	... 22 common frames omitted

2017-08-28 07:29:57.136  INFO [-,,,] 1 --- [pool-4-thread-1] i.s.c.integration.IntegrationController  : Integration -Ksbjn_kzGflCW57ed4Q : Setting status to Pending
2017-08-28 07:30:15.974  INFO [-,,,] 1 --- [pool-5-thread-1] i.s.c.integration.IntegrationController  : Integration 5 : Desired status "Activated" != current status "Pending" --> calling status change handler
```

--- a/runtime/src/main/java/io/syndesis/runtime/Application.java
+++ b/runtime/src/main/java/io/syndesis/runtime/Application.java
@@ -24,6 +24,7 @@ import io.syndesis.rest.v1.state.ClientSideState;
 import io.syndesis.rest.v1.state.ClientSideStateProperties;
 import io.syndesis.rest.v1.state.StaticEdition;
 
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -90,13 +91,16 @@ public class Application extends SpringBootServletInitializer {
     }
 
     @Bean
-    public Validator localValidatorFactoryBean(final MessageSource messageSource, final ResourcePatternResolver resolver) throws IOException {
+    public Validator localValidatorFactoryBean(final ResteasyProviderFactory factory, final MessageSource messageSource, final ResourcePatternResolver resolver) throws IOException {
         final LocalValidatorFactoryBean localValidatorFactoryBean = new LocalValidatorFactoryBean();
         localValidatorFactoryBean.setValidationMessageSource(messageSource);
 
         final Resource[] mappings = resolver.getResources("classpath*:/META-INF/validation/*.xml");
         localValidatorFactoryBean.setMappingLocations(mappings);
 
+        factory.register(new ValidatorContextResolver(localValidatorFactoryBean));
+
         return localValidatorFactoryBean;
     }
+
 }

--- a/runtime/src/main/java/io/syndesis/runtime/ValidatorContextResolver.java
+++ b/runtime/src/main/java/io/syndesis/runtime/ValidatorContextResolver.java
@@ -24,18 +24,14 @@ import javax.validation.Validation;
 import javax.validation.ValidatorFactory;
 import javax.validation.executable.ExecutableType;
 import javax.ws.rs.ext.ContextResolver;
-import javax.ws.rs.ext.Provider;
 
 import org.hibernate.validator.messageinterpolation.ResourceBundleMessageInterpolator;
 import org.hibernate.validator.resourceloading.PlatformResourceBundleLocator;
 import org.hibernate.validator.spi.resourceloading.ResourceBundleLocator;
 import org.jboss.resteasy.plugins.validation.GeneralValidatorImpl;
 import org.jboss.resteasy.spi.validation.GeneralValidator;
-import org.springframework.stereotype.Component;
 
-@Component
-@Provider
-public class ValidatorContextResolver implements ContextResolver<GeneralValidator> {
+class ValidatorContextResolver implements ContextResolver<GeneralValidator> {
 
     private final ValidatorFactory validatorFactory;
 

--- a/runtime/src/main/resources/META-INF/services/javax.ws.rs.ext.Providers
+++ b/runtime/src/main/resources/META-INF/services/javax.ws.rs.ext.Providers
@@ -1,1 +1,0 @@
-io.syndesis.runtime.ValidatorContextResolver

